packages/nodes-base/nodes/Merge/v3/actions/mode/combineBySql.ts

```diff
@@ -0,0 +1,135 @@
+import type {
+	IDataObject,
+	IExecuteFunctions,
+	INodeExecutionData,
+	INodeProperties,
+	IPairedItemData,
+} from 'n8n-workflow';
+import { NodeOperationError } from 'n8n-workflow';
+
+import { getResolvables, updateDisplayOptions } from '@utils/utilities';
+import { numberInputsProperty } from '../../helpers/descriptions';
+
+import alasql from 'alasql';
+import type { Database } from 'alasql';
+
+export const properties: INodeProperties[] = [
+	numberInputsProperty,
+	{
+		displayName: 'Query',
+		name: 'query',
+		type: 'string',
+		default: 'SELECT * FROM input1 LEFT JOIN input2 ON input1.name = input2.id',
+		noDataExpression: true,
+		description: 'Input data available as tables with coresponding number, e.g. input1, input2',
+		hint: 'Supports <a href="https://github.com/alasql/alasql/wiki/Supported-SQL-statements" target="_blank">most</a> of the SQL-99 language',
+		required: true,
+		typeOptions: {
+			rows: 5,
+			editor: 'sqlEditor',
+		},
+	},
+];
+
+const displayOptions = {
+	show: {
+		mode: ['combineBySql'],
+	},
+};
+
+export const description = updateDisplayOptions(displayOptions, properties);
+
+export async function execute(
+	this: IExecuteFunctions,
+	inputsData: INodeExecutionData[][],
+): Promise<INodeExecutionData[]> {
+	const nodeId = this.getNode().id;
+	const returnData: INodeExecutionData[] = [];
+	const pairedItem: IPairedItemData[] = [];
+
+	const db: typeof Database = new (alasql as any).Database(nodeId);
```

Comment:
could you avoid using any?

Reply:
no, there is an issue with type export in `alasql` - 'This expression is not constructable.
  Type 'Database' has no construct signatures.', so need to use any to avoid this
  
  creating database as shown in [documentation](https://github.com/AlaSQL/alasql/wiki/Create-Database):
  
![image](https://github.com/n8n-io/n8n/assets/88898367/8900c528-730b-41ce-ae68-fe74152cb976)